x-pack/test/stack_functional_integration/configs/config.stack_functional_integration_base.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resolve } from 'path';
+import buildState from './build_state';
+import { ToolingLog } from '@kbn/dev-utils';
+import chalk from 'chalk';
+import { esTestConfig, kbnTestConfig } from '@kbn/test';
+
+const reportName = 'Stack Functional Integration Tests';
+const testsFolder = '../test/functional/apps';
+const stateFilePath = '../../../../../integration-test/qa/envvars.sh';
+const prepend = (testFile) => require.resolve(`${testsFolder}/${testFile}`);
+const log = new ToolingLog({
+  level: 'info',
+  writeTo: process.stdout,
+});
+
+export default async ({ readConfigFile }) => {
+  const defaultConfigs = await readConfigFile(require.resolve('../../functional/config'));
+  // const { apps } = defaultConfigs.getAll();
+  const { tests, ...provisionedConfigs } = buildState(resolve(__dirname, stateFilePath));
+
+  const servers = {
+    kibana: kbnTestConfig.getUrlParts(),
+    elasticsearch: esTestConfig.getUrlParts(),
+  };
+  log.info(`servers data: ${JSON.stringify(servers)}`);
+  return {
+    ...defaultConfigs.getAll(),
+    junit: {
+      reportName: `${reportName} - ${provisionedConfigs.VM}`,
+    },
+    servers,
+    apps: {
+      kibana: {
```

Comment:
If we can, we should pull the apps in from test/functional/config.js and x-pack/test/functional/config.js otherwise backporting is going to be a pain.